runtime/util/resolvehelp.c

```diff
@@ -40,8 +40,12 @@ getMethodForSpecialSend(J9VMThread *vmStruct, J9Class *currentClass, J9Class *re
 	 *	C) Resolved method is not <init>
 	 *	D) Skip checking vTables if resolved or current class is an interface
 	 */
-	if ((J9AccSuper == (currentClass->romClass->modifiers & J9AccSuper))
-	|| J9_ARE_NO_BITS_SET(vmStruct->javaVM->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_ALLOW_NON_VIRTUAL_CALLS)
+	if (
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+		(J9AccSuper == (currentClass->romClass->modifiers & J9AccSuper)) ||
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```

Comment:
Please invert the test and remove `#else`:
```c
#if !defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
		(J9AccSuper == (currentClass->romClass->modifiers & J9AccSuper)) ||
#endif /* !defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```
Please also update the comments above.